File: src/domain/entities/Review.ts
import User from "./User";

export default class Review {
  constructor(
    public product_id: number,
    public rating: number,
    public user?: User,
    public title?: string,
    public content?: string,
    public verified_purchase?: boolean,
    public helpful_votes?: number,
    public id?: number,
    public created_at?: Date,
    public updated_at?: Date,
  ) {}
}

File: ecommerce/src/main/java/com/june/ecommerce/domain/review/Review.java
package com.june.ecommerce.domain.review;

import com.june.ecommerce.domain.product.Product;
import com.june.ecommerce.domain.user.User;
import jakarta.persistence.*;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Entity
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    private int rating;
    private String title;
    private String content;

    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();

    private Boolean verifiedPurchase = false;
    private Integer helpfulVotes  = 0;
}
